client/Packages/com.beamable.server/Editor/Usam/Editor/Signposts/BeamServiceImporter.cs

```diff
@@ -79,6 +79,23 @@ public override void OnInspectorGUI()
 					} while (property.NextVisible(false));
 
 				}
+				
+				if (GUILayout.Button("Apply Changes"))
+				{
+					if (_serviceAsset != null)
+					{
+						if (!_serviceAsset.data.CheckAllValidAssemblies())
+						{
+							Debug.LogError("Assembly references in file are not valid");

```

Comment:
it would be good to say WHY the file isn't valid. In the `CheckAllValidAssemblies` function, I see you're returning false in a few places... what about changing the sig to something like, 

```csharp
public static bool CheckAllValidAssemblies(out string validationMessage) {
  validationMessage = null;
  if (notValidSomeHow) { 
    validationMessage = "the eclipse flipped all the bits";
    return false;
  }
  return true;
}